pandas/_libs/src/ujson/python/objToJSON.c

```diff
@@ -329,6 +329,24 @@ static Py_ssize_t get_attr_length(PyObject *obj, char *attr) {
     return ret;
 }
 
+npy_int64 get_long_attr(PyObject *o, const char *attr) {
+  npy_int64 long_val;
+  PyObject *value = PyObject_GetAttrString(o, attr);
+  long_val = (PyLong_Check(value) ?
+              PyLong_AsLongLong(value) : PyInt_AS_LONG(value));
+  Py_DECREF(value);
+  return long_val;
+}
+
+npy_float64 total_seconds(PyObject *td) {
```

Comment:
Just pushed a comment suggesting it be removed.


> does `npy_int64 days = get_long_attr(td, "total_seconds");` work?
@jorisvandenbossche I didn't try that specifically because `total_seconds` is a nullary method.  What I tried (and failed) was several variants of `PyObject_CallMethod`